packages/@react-stately/layout/src/GridLayout.ts

```diff
@@ -223,7 +224,46 @@ export class GridLayout<T, O extends GridLayoutOptions = GridLayoutOptions> exte
     x += this.virtualizer!.visibleRect.x;
     y += this.virtualizer!.visibleRect.y;
 
-    let key = this.virtualizer!.keyAtPoint(new Point(x, y));
+    // Find the closest item within on either side of the point using the gap width.
+    let key: Key | null = null;
+    if (this.numColumns === 1) {
+      let searchRect = new Rect(x, Math.max(0, y - this.gap.height), 1, this.gap.height * 2);
+      let candidates = this.getVisibleLayoutInfos(searchRect);
+      let minDistance = Infinity;
+      for (let candidate of candidates) {
+        // Ignore items outside the search rect, e.g. persisted keys.
+        if (!candidate.rect.intersects(searchRect)) {
+          continue;
+        }
+
+        let yDist = Math.abs(candidate.rect.y - x);
+        let maxYDist = Math.abs(candidate.rect.maxY - x);
```

Comment:
maybe I'm having a brain fart and not visualizing this quite right, but shouldn't this be `candidate.rect.y - y` instead? Like comparing the candidate's top/bottom edge's Y position with the current point to find the closest edge ->  reiterating on the next candidate to find the candidate that is closest to the drop point?

Reply:
https://github.com/adobe/react-spectrum/pull/7893